scanpy/tests/test_score_genes.py

```diff
@@ -43,6 +47,25 @@ def _create_adata(n_obs, n_var, p_zero, p_nan):
     return adata
 
 
+def test_score_with_reference():
+    """
+    Checks if score_genes output agrees with pre-computed reference values.
+    The reference values had been generated using the same code
+    and stored as a pickle object in ./data
+    """
+
+    np.random.seed(123)
+    adata = sc.datasets.paul15()
+    sc.pp.normalize_per_cell(adata, counts_per_cell_after=10000)
+    sc.pp.scale(adata)
+
+    sc.tl.score_genes(adata, gene_list=adata.var_names[:100], score_name='Test')
```

Comment:
What is the `np.random.seed(123)` call for?

I don't think there should be any non-determinism in the setup here. Also, this shouldn't have an effect on `sc.tl.score_genes` since randomness there should be controlled by the `random_state` parameter which defaults to 0.

As it stands, removing the `np.random.seed(123)` call shoudn't have any effect on this test. Did you want to do something more like this?

```suggestion
    adata = sc.datasets.paul15()
    sc.pp.normalize_per_cell(adata, counts_per_cell_after=10000)
    sc.pp.scale(adata)

    sc.tl.score_genes(adata, gene_list=adata.var_names[:100], score_name='Test', random_state=123)
```

This would change the output though.

Reply:
It's a leftover line from the previous test which used randomly generated integers, I just forgot to remove it. There is no need for specifying the seed. Test works fine after removing the line.
The reference values were calculated with default sc.tl.score_genes (=0).